pkg/resourcemonitor/resourcemonitor.go

```diff
@@ -273,6 +285,14 @@ func GetAllContainerDevices(podRes []*podresourcesapi.PodResources, namespace st
 	return allCntRes
 }
 
+func ComputePodFingerprint(podRes []*podresourcesapi.PodResources) string {
+	fp := podfingerprint.NewFingerprint(len(podRes))
+	for _, pr := range podRes {
+		fp.AddPod(pr)
+	}
+	return fp.Sign()
```

Comment:
this is a great conversation to have as a issue of the https://github.com/k8stopologyawareschedwg/podfingerprint because it's a (very good) observation about how the fingerprinting is done; here we "just" care about getting the value from a black box and relaying it properly :)

In a nutshell, what you say is just true. The pod UID would be a much better representation of the pod identity, being guaranteed to be unique. The pod UID should also provide stronger guarantees when Vertical Pod Autoscaler lands.

The problem is extracting the UID. From the podresources API we don't have this information (this is arguably a oversight), and accessing the local kubelet endpoint is doable and a good option but it has its own set of complication in openshift (need to explicitely enable clients) and in general, because the updater agent would need to reconcile two different sources of truth, with all the extra challenges and its own set of corner cases. For example, what if the running pod set changes between the calls to podresources List and listing the pods through the kubelet local endpoint? Is this more or less likely than the scenario we are talking about?

Reply:
(Capturing some thoughts based on an offline discussion we had about this topic)
For catch Invalidation checks, false negatives are still okay but we need to make sure we avoid false positives. Reconciling the two states of truth brings with itself another set of challenges indeed. An alternative could be to take resources into consideration while evaluating the fingerprint as the cache updates made by the reserve plugin are based on the resources requested by the pod. We can be more certain about the pod uniqueness if the fingerprint is a function of name, namespace and resources.

Reply:
yes, we should file an issue into the podfingerprint repo and move the discussion there.
Additional question is: is this issue bad enough to block this PR?

Reply:
> An alternative could be to take resources into consideration while evaluating the fingerprint as the cache updates made by the reserve plugin are based on the resources requested by the pod. 

I agree this is a good middle-ground for solving the issue.
But I thought it's already what the fingerprint algorithm is doing

Reply:
I don't think it is blocker for this PR. This is a corner case and most likely (fingers crossed) we won't run into it. I can file an issue in the podfingerprint repo so it does not get lost. Acknowledging that this could be an issue and there is room for improvement is probably the best we can do now.

Reply:
> I agree. This is a good compromise for handling the issue. But I thought it's already what the fingerprint algorithm is doing
Currently the fingerprint evaluation only takes name and namespace into consideration and not resources.

Reply:
We surely need to acknowledge and document this issue! the reason why we should be able to mitigate this problem is because it """only""" manifests itself when we reuse the pod name for different pods. But most of the pods are (or should be) run by controllers, and controllers (deployment, daemonsets...) should give unique names to pod they manage anyway.
So, while again very important to acknowledge this limitation, the practical impact should be manageable.